### PR TITLE
fix #4101: select a random visible blog if the primary blog is not available

### LINKED ISSUE #4101: Removed Jetpack blog reappears after signing in
I can repro these steps on both an emulator and a real device (N6P):
* Go to the settings of a JP site and remove it from the app
* Disconnect from wp.com
* Login again
* The JP site now shows up again and is the default blog

The problem appears to be due to [this line](https://github.com/wordpress-mobile/WordPress-Android/blob/80246ad8abc5477372dfe1425428b2e5858ecd03/WordPress/src/main/java/org/wordpress/android/WordPress.java#L684) in `attemptToRestoreLastActiveBlog`. That call to `getFirstVisibleBlogId` returns the ID of my JP site.

### PR DESCRIPTION
**Pull request by maxme**
fix #4101: select a random visible blog if the primary blog is not available

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 1]
```diff
@@ -534,17 +534,28 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
         }
     };
 
+    private void refreshAndSelectSite(Map<String, Object> site) {
+        Map<String, Object> primaryBlog = site;
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: The variable `primaryBlog` is redundant here.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 2]
```diff
@@ -534,17 +534,28 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
...
-                Map<String, Object> primaryBlog = blogs.get(0);
-                // Ask for a refresh and select it
-                refreshBlogContent(primaryBlog);
-                WordPress.setCurrentBlog((Integer) primaryBlog.get("id"));
+            if (sites != null && !sites.isEmpty()) {
+                refreshAndSelectSite(sites.get(0));
+            } else {
+                // Primary blog not found or hidden (can happen if it's a Jetpack site)
+                // Select the first visible site if it exists
+                sites = WordPress.wpDB.getBlogsBy("isHidden = 0", null, 1, true);
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: Should be ` WordPress.wpDB.getBlogsBy("isHidden = 0", null, 1, false);` here

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java (hunk 1 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: Fixed in 0920d44

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**